modin/core/io/text/text_file_dispatcher.py

```diff
@@ -1115,3 +1113,44 @@ def _get_skip_mask(cls, rows_index: pandas.Index, skiprows: Callable):
             mask = rows_index.map(skiprows)
 
         return mask
+
+    @classmethod
+    @logger_decorator("PANDAS-API", "TextFileDispatcher._should_pass_names", "INFO")
+    def _should_pass_names(cls, names, skiprows, skipfooter, usecols):
```

Comment:
```suggestion
    def _whether_to_pass_names(cls, names, skiprows, skipfooter, usecols):
```
I wonder if this is clear/better?

Reply:
I vote for `_should_pass_names`. It seems simpler. If I'm calling this method I would think to myself, "Should I pass names?" and the method will tell me.

Reply:
It seems to me `_should_pass_names` tells/recommends us to pass names, whereas `_whether_to_pass_names` really asks if we pass names or not. Does this make sense?

Reply:
Hm, that's not what it sounds like to me. Given that we invoke the function, it seems clear that we are trying to figure out whether to pass names. The code

```python
if _should_pass_names(names, skiprows, skipfooter, usecols):
  pass_names
```

in my head is like "if should pass names, pass names" which is pretty close to the real English interpretation which is "if we should pass names according to the function, pass names". whereas `_whether_to_pass_names` would sound like "if whether to pass names, pass names", which is more confusing.

Reply:
Given the signature of the `_should_pass_names` function, it is not clear to me that we should use it in a condition. Let's allow others to weigh in this too.

Reply:
I have changed the function name (with additional changes of if else in some places of `_read`) to make its responsibilities more clear. For now, the name `_should_use_inferred_columns_names` can be interpreted definitely, as for me _Should we use inferred columns names or not_.


@mvashishtha, @YarShev, thoughts?

Reply:
I think that name is clear.

Reply:
Functions that return True/False tend to be named using prefix `is` so I suggest renaming this function to `_is_use_inferred_columns_names`.

Reply:
I think that bool functions should start with `is` if and only if they are describing an object. For example, `pandas.Index.is_unique` is describing the index. 

```python
if index.is_unique:
  foo()
```

is "if index is unique, `foo()`", which sounds a lot like the full sentence "if the index is unique, do `foo()`".

Here `use_inferred_columns_names` is not describing something. It's telling us whether to take the action of using inferred column names." If I write 

```python
if is_use_inferred_columns_names:
  pass_names()
```

it sounds like "if is use inferred columns names, pass names." On the other hand, if we had `use_inferred_columns_names`, the code would sound like "If use inferred columns names, pass names." That sounds much more natural to me.

Reply:
> I think that bool functions should start with `is` if and only if they are describing an object.

Not sure of that. You can look at the NumPy logic functions [link](https://numpy.org/doc/stable/reference/routines.logic.html), for instance. They take a NumPy array as an input though. We should rely on the conventional naming for the functions/variables/etc. and I am still not convinced to use `_should_use_inferred_column_names`.

Also, considering your point about bool functions that should start with `is` if and only if they are describing an object we use `cls._should_use_inferred_column_names` with the class instead of using this method as a free function.

Maybe `is_inferred_columns_names_used` sounds better and clearer to you?

Reply:
> You can look at the NumPy logic functions [link](https://numpy.org/doc/stable/reference/routines.logic.html), for instance.

I don't understand this point. The `is` functions I think you are referring to are each telling whether a particular descriptor matches an object. e.g. `isreal` tells whether the object is real. In this case the function is telling whether, given the `read_csv` arguments, the dispatcher "should use inferred columns". You could say that "should use inferred columns" is itself a descriptor, but I think it's more like a directive, telling the dispatcher what to do. I actually think that it **should** be a free function because it's not using any data from the `cls`. 

`is_inferred_columns_names_used` is ok with me but seems less clear than `use_inferred_columns_names` or `should_use_inferred_columns_names`. Generally for writing it's clearer to use the active voice (see e.g. [here](https://developers.google.com/tech-writing/one/active-voice)) and I think that applies to code too.

Reply:
> I actually think that it **should** be a free function because it's not using any data from the `cls`.

Good point! We can make it a static method as the function is tied to the class logic.

I am getting convinced now to use `use_inferred_columns_names`/`should_use_inferred_columns_names` or even `uses_inferred_columns_names`, which tells us if the class uses inferred column names in workers or not. Which function makes much more sense to you?

Reply:
`staticmethod` was applied. The method was renamed to `_uses_inferred_columns_names`.